plugins/message-markdown-preview/src/components/PreviewButton.tsx

```diff
@@ -10,126 +10,78 @@ import { General } from "@vendetta/ui/components";
 import { semanticColors } from "@vendetta/ui";
 import { FadeView } from "../../../../stuff/animations";
 import { findByName, findByProps, findByStoreName } from "@vendetta/metro";
-import { before, instead } from "@vendetta/patcher";
+import { after } from "@vendetta/patcher";
 
-const { Pressable } = General;
+const { ScrollView, Pressable } = General;
 
-const { default: renderMessageMarkup } = findByProps(
-  "renderMessageMarkupToAST"
-);
-const MessagePreview = findByName("MessagePreview", false);
-const UserStore = findByStoreName("UserStore");
-const SelectedChannelStore = findByStoreName("SelectedChannelStore");
-
-export default ({
-  thing,
-}: {
-  thing: { runner: (txt: string) => void };
-}): React.JSX.Element => {
-  const size = 40;
-  const styles = stylesheet.createThemedStyleSheet({
-    androidRipple: {
-      color: semanticColors.ANDROID_RIPPLE,
-      cornerRadius: 2147483647,
-    },
-    actionButton: {
-      borderRadius: 2147483647,
-      height: size,
-      width: size,
-      marginHorizontal: 4,
-      flexShrink: 0,
-      flexDirection: "row",
-      alignItems: "center",
-      justifyContent: "center",
-      backgroundColor: semanticColors.BACKGROUND_SECONDARY,
+const ACTION_ICON_SIZE = 40;
+const styles = stylesheet.createThemedStyleSheet({
+  androidRipple: {
+    color: semanticColors.ANDROID_RIPPLE,
+    cornerRadius: 2147483647,
+  },
+  actionButton: {
+    borderRadius: 2147483647,
+    height: ACTION_ICON_SIZE,
+    width: ACTION_ICON_SIZE,
+    marginHorizontal: 4,
+    flexShrink: 0,
+    flexDirection: "row",
+    alignItems: "center",
+    justifyContent: "center",
+    backgroundColor: semanticColors.BACKGROUND_SECONDARY,
 
-      marginLeft: 8,
-      marginTop: -4,
-    },
-    actionIcon: {
-      tintColor: semanticColors.INTERACTIVE_NORMAL,
-      width: size * 0.6,
-      height: size * 0.6,
-    },
-  });
+    marginLeft: 8,
+    marginTop: -4,
+  },
+  actionIcon: {
+    tintColor: semanticColors.INTERACTIVE_NORMAL,
+    width: ACTION_ICON_SIZE * 0.6,
+    height: ACTION_ICON_SIZE * 0.6,
+  },
+});
 
-  const [text, setText] = React.useState("");
-  thing.runner = (txt) => setText(txt);
+const { default: ChatItemWrapper } = findByProps("DCDAutoModerationSystemMessageView");
+const MessageRecord = findByName("MessageRecord");
+const RowManager = findByName("RowManager");
+const SelectedChannelStore = findByStoreName("SelectedChannelStore");
+const UserStore = findByStoreName("UserStore");
 
-  const run = () => {
-    const markup = renderMessageMarkup(
-      {
-        content: text,
-        embeds: [],
-      },
-      {
-        hideSimpleEmbedContent: true,
-        formatInline: false,
-        allowHeading: true,
-        allowList: true,
-        allowLinks: false,
-        previewLinkTarget: false,
-      }
-    ).content;
+export default ({ inputProps }): JSX.Element => {
+  const textRef = React.useRef<string>(null);
```

Comment:
wouldn't it be easier to use a state

Reply:
We don't really need the component to re-render, hence the `useRef` *(we don't necessarily need to use `useRef` either, actually)*